### PR TITLE
fix spawn with empty magazine loaded - Mag Well system

### PR DESCRIPTION
**Pull request by NikPoliakov**
**When merged this pull request will:**
- Fixed spawn with empty magazines loaded in the primary weapon.
- Added getMagwellMagazines function that returns array of compatible magazines.

### CODE REVIEW

File: cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf [hunk 1]
```diff
@@ -8,9 +8,10 @@ if (_weapons isEqualTo []) then {
 
 private _weapon = _weapons deleteAt 0;
 private _loadedMagazines = _magazines deleteAt 0;
-private _compatibleMagsFirstMuzzle = getArray (configfile >> "CfgWeapons" >> _weapon >> "magazines");
-private _compatibleMagsSecondMuzzle = [];
 
+_compatibleMagsFirstMuzzle = [_weapon] call FUNC(getMagwellMagazines);
+
+private _compatibleMagsSecondMuzzle = [];
```
Comment (ColdEvul) [inline cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf:14]: private declared after usage should be swapped

File: cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf [hunk 2]
```diff
@@ -8,9 +8,10 @@ if (_weapons isEqualTo []) then {
 
 private _weapon = _weapons deleteAt 0;
 private _loadedMagazines = _magazines deleteAt 0;
-private _compatibleMagsFirstMuzzle = getArray (configfile >> "CfgWeapons" >> _weapon >> "magazines");
-private _compatibleMagsSecondMuzzle = [];
 
+_compatibleMagsFirstMuzzle = [_weapon] call FUNC(getMagwellMagazines);
```
Comment (ColdEvul) [inline cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf:12]: add `private`

File: cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf [hunk 3]
```diff
@@ -8,9 +8,10 @@ if (_weapons isEqualTo []) then {
...
+
+private _compatibleMagsSecondMuzzle = [];
 private _secondMuzzle = (getArray (configfile >> "CfgWeapons" >> _weapon >> "muzzles") - ["this"]) param [0, ""];
 if (_secondMuzzle != "") then {
     _compatibleMagsSecondMuzzle = getArray (configfile >> "CfgWeapons" >> _weapon >> _secondMuzzle >> "magazines");
```
Comment (ColdEvul) [inline cScripts/Loadouts/script/functions/main/fn_findLoadableWeapon.sqf:17]: This also checks same and should also be calling the function

**Comment by ColdEvul** [comment]
is `if (_muzzle != "SAFE") then {`
a value used or is it a value you made?

**Comment by NikPoliakov** [comment]
every weapon has a SAFE muzzle.

**Comment by ColdEvul** [comment]
Aight looks good!
You cold do some indent cleaning cause its usually 4 we use you mixed there :)

**Pull request merged**